File: 0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=nums[0];
        int temp_sum=0;
        int start=0;
        while(start<nums.length){
          temp_sum+=nums[start];
            
            if(temp_sum > sum){
                sum=temp_sum;
            }
            
            if(temp_sum<=0){
               temp_sum=0;
            }
                
            start++;
        }
            return sum;
    }
}

File: 0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSubarraySum = nums[0];
        int tempSum = 0;
        for(int x : nums){
            tempSum += x;

            if(tempSum > maxSubarraySum){
                maxSubarraySum = tempSum;
            }

            if(tempSum <= 0) tempSum = 0;
        }
        return maxSubarraySum;
    }
};
